velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -129,25 +130,25 @@ class AverageAggregate : public exec::Aggregate {
 
     if (decodedRaw_.isConstantMapping()) {
       if (!decodedRaw_.isNullAt(0)) {
-        const T value = decodedRaw_.valueAt<T>(0);
+        const TInput value = decodedRaw_.valueAt<TInput>(0);
         const auto numRows = rows.countSelected();
         updateNonNullValue(group, numRows, value * numRows);
       }
     } else if (decodedRaw_.mayHaveNulls()) {
       rows.applyToSelected([&](vector_size_t i) {
         if (!decodedRaw_.isNullAt(i)) {
-          updateNonNullValue(group, decodedRaw_.valueAt<T>(i));
+          updateNonNullValue(group, decodedRaw_.valueAt<TInput>(i));
         }
       });
     } else if (!exec::Aggregate::numNulls_ && decodedRaw_.isIdentityMapping()) {
-      const T* data = decodedRaw_.data<T>();
+      const TInput* data = decodedRaw_.data<TInput>();
       double totalSum = 0;
```

Comment:
should this be `TAccumulator totalSum`?

Reply:
Fixed in this PR: Some of these are fixed in this PR: https://github.com/facebookincubator/velox/pull/2639/files#diff-013ebdfa6873d609e447826a2201cbf5e1a91c9f509cbe8f65452b5959cf2ad5R247

These issues get highlighted when I specialize for Decimals.

Reply:
I see. It will be nice to keep this PR complete in terms of templating. When we add decimal support, it should be enough to add only the registration.